src/toil_scripts/gatk_germline/germline.py

```diff
@@ -1,688 +1,570 @@
+                            name='toil.1.fq',
+                            s3_key_path=config.ssec,
+                            disk='50G')
+    # Assumes second fastq url is identical
+    fq2_url = url.replace('1.fq', '2.fq')
+    fq2 = job.addChildJobFn(download_url_job,
+                            fq2_url,
+                            name='toil.2.fq',
+                            s3_key_path=config.ssec,
+                            disk='50G')
+
+    bwa_config = deepcopy(config)
+    bwa_config.r1 = fq1.rv()
+    bwa_config.r2 = fq2.rv()
+    # bwa_alignment uses a different naming convention
+    bwa_config.ref = config.genome_fasta
+    bwa_config.fai = config.genome_fai
+    bwakit_disk = PromisedRequirement(lambda x, y: 3*(x.size + y.size) + human2bytes('8G'),
+                                      fq1.rv(), fq2.rv())
+    return job.addFollowOnJobFn(run_bwakit,
+                                bwa_config,
+                                cores,
+                                sort=False,
+                                trim=config.trim,
+                                cores=cores,
+                                disk=bwakit_disk).rv()
+
+
+def gatk_haplotype_caller(job, bam_id, bai_id, config, emit_threshold=10.0, call_threshold=30.0):
     """
     Uses GATK HaplotypeCaller to identify SNPs and Indels and writes a gVCF.
-    Calls per-sample genotyper to genotype gVCF.
 
-    :param job: Job instance
-    :param shared_ids: dictionary of shared file promises
-    :param input_args: dictionary of input arguments
-    """
+    :param JobFunctionWrappingJob job: passed automatically by Toil
+    :param str bam_id: BAM FileStoreID
+    :param str bai_id: BAI FileStoreID
+    :param Namespace config: pipeline configuration options and shared files
+    :param float emit_threshold: Minimum phred-scale confidence threshold for a variant to be emitted
+                               Default: 10
+    :param float call_threshold: Minimum phred-scale confidence threshold for a variant to be called
+                               Default: 30
+    :return: GVCF FileStoreID
+    :rtype: str
+    """
+    job.fileStore.logToMaster('Running GATK HaplotypeCaller: {}'.format(config.uuid))
     work_dir = job.fileStore.getLocalTempDir()
-    input_files = ['ref.fa', 'ref.fa.fai', 'ref.dict', 'toil.bam', 'toil.bam.bai']
-    read_from_filestore_hc(job, work_dir, shared_ids, *input_files)
-    output = '%s.raw.BOTH%s.gvcf' % (input_args.uuid, input_args.suffix)
-
-    # Call GATK -- HaplotypeCaller
-    command = ['-U', 'ALLOW_SEQ_DICT_INCOMPATIBILITY', # RISKY! (?) See #189
-               '-nct', str(input_args.cpu_count),
-               '-R', 'ref.fa',
+    inputs = {'genome.fa': config.genome_fasta, 'genome.fa.fai': config.genome_fai,
+              'genome.dict': config.genome_dict, 'input.bam': bam_id, 'input.bam.bai': bai_id}
+    get_files_from_filestore(job, work_dir, inputs)
+
+    cores = multiprocessing.cpu_count()
```

Comment:
`cores = multiprocessing.cpu_count()` is called many times, why not call it once in the first job function?  Is this to accommodate heterogenous clusters? If so, it won't really work because the number of cores being passed as the requirement for this job function will be different than what you reset cores to when passing options to the tool.